source/hp/fe_collection.cc

```diff
@@ -160,9 +160,25 @@ namespace hp
 
 
 
+  template <int dim, int spacedim>
+  FECollection<dim, spacedim>::FECollection()
+  {
+    // establish hierarchy corresponding to indices in this collection
+    set_hierarchy(
+      [this](const unsigned int fe_index) {
+        return ((fe_index + 1) < this->size()) ? fe_index + 1 : fe_index;
```

Comment:
You should document what happens if you're already at the first (last) element of the hierarchy.